qemu/tests/numa_prealloc_handling.py

```diff
@@ -0,0 +1,29 @@
+import re
+from avocado.utils import process
+
+
+def get_qemu_exec_time(cmd_line):
+    result = process.run(cmd_line, ignore_status=True, shell=True)
+    result = str(result.stdout).split("\\t")[-1].split("\\n")[0]
+    execution_time = int(re.sub("\\D", "", result))
+    return execution_time
+
+
+def run(test, params, env):
+    """
+    numa_prealloc_handling test
+    1) Measures the time takes QEMU to preallocate the memory
+    2) Checks the timing is shorter when thread-context is used
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+    cmd_without_tc = params.get("cmd_without_tc")
+    cmd_with_tc = params.get("cmd_with_tc")
+
+    execution_time = get_qemu_exec_time(cmd_without_tc)
+
+    execution_time_tc = get_qemu_exec_time(cmd_with_tc)
+
+    if execution_time <= execution_time_tc:
```

Comment:
Comparing time strings is not a good idea, I would suggest using datetime.
```python
>>> from datetime import datetime
>>> date1 = datetime.strptime('0m1.125s', '%Mm%S.%fs')
>>> date2 = datetime.strptime('1m0.125s', '%Mm%S.%fs')
>>> date1 < date2
True
```

Reply:
Even if you convert it to integers, you lose the time unit. So if time is `1m2.100s` and `11m2.00s`, which one is bigger if using your method? I know this will not happen, but we have a better choice, why not use that?

Or if you don't want to use the python time method, the `time` command also supports return seconds output.

```
   The format string
       The format is interpreted in the usual printf-like way.  Ordinary characters are directly copied, tab, newline, and backslash are escaped using \t, \n, and \\, a percent sign is represented by %%, and otherwise %  indi‐
       cates a conversion.  The program time will always add a trailing newline itself.  The conversions follow.  All of those used by tcsh(1) are supported.

       Time

       %E     Elapsed real time (in [hours:]minutes:seconds).

       %e     (Not in tcsh(1).)  Elapsed real time (in seconds).
```
```
# /usr/bin/time -f %e sleep 1
1.00
```

Reply:
@PaulYuuu Got your idea, in that case changing to datetime.

Reply:
Updated, thanks

Reply:
I prefer the solution that @PaulYuuu mentioned to specify the output format with `/usr/bin/time`.    In this way,  
1. It does not depends on the default format of `time`  if there are any difference cross platform
2. No need to convert time

Reply:
```
# /usr/bin/time -f '%e' taskset -c 0 /usr/libexec/qemu-kvm -nographic -sandbox on,resourcecontrol=deny -monitor stdio -cpu host -object memory-backend-ram,id=mem0,size=20G,prealloc=on,prealloc-threads=4   2>&1 | tail -1
4.24
```

Highlight one thing,  the `/usr/bin/time` command is provides by the time rpm

Reply:
@yanan-fu thanks for the highlight this up, it reduces the complexity :grinning:

Reply:
suggest add:
```
            if not utils_package.package_install("time"):
                test.cancel("******")
``` 

I just noticed the `time` package is not installed by a minimal installation.
This is also a disadvantage when use `/usr/bin/time` instead of  `time`.
But the risk of install `time` failed is very low.

@PaulYuuu  Any thought?

Reply:
Included, thanks !